File: php-sdk/test/Integration/VobizIntegrationTest.php
<?php
/**
 * Vobiz PHP SDK - Integration Tests (Read-Only)
 * Run: php test/Integration/VobizIntegrationTest.php
 * Requires: VOBIZ_AUTH_ID and VOBIZ_AUTH_TOKEN env vars
 */

require_once __DIR__ . '/../../vendor/autoload.php';

use Vobiz\Configuration;
use Vobiz\ApiException;
use Vobiz\VobizApi\AccountApi;
use Vobiz\VobizApi\CallApi;
use Vobiz\VobizApi\RecordingApi;
use Vobiz\VobizApi\ConferenceApi;
use Vobiz\VobizApi\ApplicationApi;

$authId    = getenv('VOBIZ_AUTH_ID');
$authToken = getenv('VOBIZ_AUTH_TOKEN');

if (!$authId || !$authToken) {
    echo "SKIP: VOBIZ_AUTH_ID or VOBIZ_AUTH_TOKEN not set\n";
    exit(0);
}

$config = new Configuration();
$config->setApiKey('X-Auth-ID', $authId);
$config->setApiKey('X-Auth-Token', $authToken);

$passed = 0;
$failed = 0;

function runTest(string $name, callable $fn): void {
    global $passed, $failed;
    try {
        $fn();
        echo "[PHP] PASS: $name\n";
        $passed++;
    } catch (ApiException $e) {
        echo "[PHP] FAIL: $name - HTTP {$e->getCode()}: {$e->getMessage()}\n";
        $failed++;
    } catch (Exception $e) {
        echo "[PHP] FAIL: $name - {$e->getMessage()}\n";
        $failed++;
    }
}

// Test 1: Get Account Details
runTest('GetAccountDetails', function() use ($config) {
    $api = new AccountApi(null, $config);
    $result = $api->apiV1AuthMeGet();
    echo "  -> Response received OK\n";
});

// Test 2: Get Live Calls
runTest('GetLiveCalls', function() use ($config, $authId) {
    $api = new CallApi(null, $config);
    $result = $api->apiV1AccountAuthIdCallGet($authId, 'live');
    echo "  -> Response received OK\n";
});

// Test 3: List Recordings
runTest('ListRecordings', function() use ($config, $authId) {
    $api = new RecordingApi(null, $config);
    $result = $api->apiV1AccountAccountIdRecordingGet($authId, 20, 0);
    echo "  -> Response received OK\n";
});

// Test 4: List Conferences
runTest('ListConferences', function() use ($config, $authId) {
    $api = new ConferenceApi(null, $config);
    $result = $api->apiV1AccountAuthIdConferenceGet($authId);
    echo "  -> Response received OK\n";
});

// Test 5: List Applications
runTest('ListApplications', function() use ($config, $authId) {
    $api = new ApplicationApi(null, $config);
    $result = $api->apiV1AccountAuthIdApplicationGet($authId);
    echo "  -> Response received OK\n";
});

echo "\n[PHP] Results: $passed passed, $failed failed\n";
exit($failed > 0 ? 1 : 0);

File: java-sdk/src/test/java/ai/vobiz/api/VobizIntegrationTest.java
package ai.vobiz.api;

import ai.vobiz.ApiClient;
import ai.vobiz.ApiException;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;

import static org.junit.jupiter.api.Assertions.*;

/**
 * Vobiz Java SDK - Integration Tests (Read-Only)
 * Requires VOBIZ_AUTH_ID and VOBIZ_AUTH_TOKEN environment variables.
 */
@EnabledIfEnvironmentVariable(named = "VOBIZ_AUTH_ID", matches = ".+")
public class VobizIntegrationTest {

    private static ApiClient client;
    private static String authId;

    @BeforeAll
    static void setup() {
        authId = System.getenv("VOBIZ_AUTH_ID");
        String authToken = System.getenv("VOBIZ_AUTH_TOKEN");

        client = new ApiClient();
        client.addDefaultHeader("X-Auth-ID", authId);
        client.addDefaultHeader("X-Auth-Token", authToken);
    }

    @Test
    void testGetAccountDetails() throws ApiException {
        AccountApi api = new AccountApi(client);
        Object result = api.apiV1AuthMeGet();
        System.out.println("[Java] GetAccountDetails: OK");
        assertNotNull(result);
    }

    @Test
    void testGetLiveCalls() throws ApiException {
        CallApi api = new CallApi(client);
        Object result = api.apiV1AccountAuthIdCallGet(authId, "live");
        System.out.println("[Java] GetLiveCalls: OK");
        assertNotNull(result);
    }

    @Test
    void testListRecordings() throws ApiException {
        RecordingApi api = new RecordingApi(client);
        Object result = api.apiV1AccountAccountIdRecordingGet(authId, 20, 0, null, null);
        System.out.println("[Java] ListRecordings: OK");
        assertNotNull(result);
    }

    @Test
    void testListConferences() throws ApiException {
        ConferenceApi api = new ConferenceApi(client);
        Object result = api.apiV1AccountAuthIdConferenceGet(authId);
        System.out.println("[Java] ListConferences: OK");
        assertNotNull(result);
    }

    @Test
    void testListApplications() throws ApiException {
        ApplicationApi api = new ApplicationApi(client);
        Object result = api.apiV1AccountAuthIdApplicationGet(authId, 20, 0);
        System.out.println("[Java] ListApplications: OK");
        assertNotNull(result);
    }
}
